platform/keyword-registry/src/main/java/org/codice/ddf/keyword/KeywordRegistryImpl.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.keyword;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ResourceBundle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeywordRegistryImpl implements KeywordRegistry {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KeywordRegistryImpl.class);
+
+  private Map<String, String> keywords = new HashMap<>();
+
+  private static final String DEFAULT_LOCALE = "KeywordRegistryBundle";
+
+  @Override
+  public Optional<String> getKeyword(String keyword) {
+    return Optional.ofNullable(keywords.get(keyword));
+  }
+
+  @Override
+  public Map<String, String> getKeywords() {
+    return ImmutableMap.copyOf(keywords);
+  }
+
+  @SuppressWarnings("WeakerAccess" /* public method needed for blueprint/metatype */)
+  public void loadDefaultLocale() {
```

Comment:
LOCALE.ENGLISH is not necessarily the default right? The default would be whatever the system is configured for or is specified https://www.oracle.com/technetwork/articles/javase/locale-140624.html#using. Should we name this method loadEnglishLocale()? Should that be an API method or should we attempt to use the DefaultLocale and fallback to an English Locale if the default doesnt exist? (e.g. fr_CA isn't there).

Reply:
You can always provide a default one that will be a catch all for all locales. The file would simply not have any locale name suffixed to it. At a previous company, we would always make sure that for any resource bundles, a default one must be specified and contains the english translations of everything. Then you override what you need to in language and/or language&country specific files.